spark/src/main/scala/org/apache/spark/sql/delta/coordinatedcommits/UCCommitCoordinatorBuilder.scala

```
@@ -240,15 +257,90 @@ object UCCommitCoordinatorBuilder
-  def createUCClient(uri: String, token: String): UCClient
+  def createUCClient(uri: String, token: String): UCClient =
+    createUCClient(uri, new FixedUCTokenProvider(token))
+
+  def createUCClient(uri: String, provider: UCTokenProvider): UCClient
 }
 
 object UCTokenBasedRestClientFactory extends UCClientFactory {
-  override def createUCClient(uri: String, token: String): UCClient =
-    new UCTokenBasedRestClient(uri, token)
+  override def createUCClient(uri: String, provider: UCTokenProvider): UCClient =
+    new UCTokenBasedRestClient(uri, provider)
+}
+
+case class UCClientParams(
+    uri: Option[String],
+    token: Option[String] = None,
+    oauthUri: Option[String] = None,
+    oauthClientId: Option[String] = None,
+    oauthClientSecret: Option[String] = None) {
+  def buildUCClient(ucClientFactory: UCClientFactory): UCClient = {
+    (uri, token, oauthUri, oauthClientId, oauthClientSecret) match {
+      case (Some(u), Some(t), _, _, _) =>
+        ucClientFactory.createUCClient(u, t)
+      case (Some(u), _, Some(oUri), Some(oClientId), Some(oClientSecret)) =>
+        val provider = new OAuthUCTokenProvider(oUri, oClientId, oClientSecret)
+        ucClientFactory.createUCClient(u, provider)
+      case _ =>
+        throw new IllegalStateException(
+          "Invalid UCClientParams, missing token or oauth credentials")
+    }
+  }
+}
+
+object UCClientParams extends DeltaLogging {
+  def create(
+      catalogName: String,
+      uri: Option[String],
+      token: Option[String] = None,
+      oauthUri: Option[String] = None,
+      oauthClientId: Option[String] = None,
+      oauthClientSecret: Option[String] = None): Option[UCClientParams] = {
+    // Validate the uri.
+    uri match {
+      case Some(u) =>
+        if (!isValidURI(u)) {
+          logWarning(log"Skipping catalog ${MDC(DeltaLogKeys.CATALOG, catalogName)} as it " +
+            log"does not have a valid URI ${MDC(DeltaLogKeys.URI, u)}.")
+          return None
```

shouldnt we just throw an error if this is the case, instead of return None?